pennylane/transforms/decompose.py

```diff
@@ -133,6 +205,12 @@ def stopping_condition(self, op: qml.operation.Operator) -> bool:
                 that the operator does not need to be decomposed.
             """
 
+            print("op: ", op)
+            if (self._fixed_decomps and isinstance(op, tuple(self._fixed_decomps.keys()))) or (
+                self._alt_decomps and isinstance(op, tuple(self._alt_decomps.keys()))
+            ):
+                return False
```

Comment:
I don't think `fixed_decomps` or `alt_decomps` should force an operator to be decomposed if it's in the target gate set. The stopping condition / gate set should be the source of truth for whether something should be decomposed or not. `fixed_decomps` and `alt_decomps` specifies **how** an operator is to be decomposed, not **if** an operator should be decomposed.